src/Node/Expression/Binary/BinaryInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression\Binary;
+
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Node\Node;
+
+/**
+ * @internal
+ */
+interface BinaryInterface
+{
+    /**
+     * @param AbstractExpression $left
+     * @param AbstractExpression $right
```

Comment:
I think the interface should define `AbstractExpression` instead. Variance rules will allow some classes to use `Node` for BC reasons (as it is a wider type)

Reply:
Updated